Snowplow/Internal/Events/SPConsentDocument.m

```diff
@@ -27,69 +27,34 @@
 #import "SPPayload.h"
 #import "SPSelfDescribingJson.h"
 
-@implementation SPConsentDocument {
-    NSString * _documentId;
-    NSString * _version;
-    NSString * _name;
-    NSString * _documentDescription;
-}
+@interface SPConsentDocument ()
 
-+ (instancetype)build:(void(^)(id<SPConsentDocumentBuilder> builder))buildBlock {
-    SPConsentDocument* document = [SPConsentDocument new];
-    if (buildBlock) { buildBlock(document); }
-    [document preconditions];
-    return document;
-}
+@property (nonatomic, readwrite) NSString *documentId;
+@property (nonatomic, readwrite) NSString *version;
 
-- (instancetype)init {
-    self = [super init];
-    return self;
-}
+@end
+
+@implementation SPConsentDocument
 
 - (instancetype)initWithDocumentId:(NSString *)documentId version:(NSString *)version {
     if (self = [super init]) {
         _documentId = documentId;
         _version = version;
+        [SPUtilities checkArgument:(_documentId != nil) withMessage:@"Document ID cannot be nil."];
```

Comment:
Ok, I know what happen here. I put the comments "//$ added" as reminder to refactor them before the PR. Then I forgot to clean them.

My plan was to avoid the use of those "preconditions" function, in fact the only events that still use them are the ones I annotated with "//$ added".

Why remove "preconditions": 
It's mostly matter of taste here. IMO `preconditions` does very little things here. It's used only in the constructor. The constructor is small and essentially it's only needed to initialise private properties.
For these reasons I think it's a fragmentation not particularly useful. The purpose of the function is clear even if those lines are integrated in the constructor. The readability isn't compromised. The testing is not compromised as well. The nullability annotation should solve many of the `preconditions` checks (even if ObjC is not great to raise errors about this - hence I'm a bit lazy to remove nullability checks in the constructor!).
My feeling is that `preconditions` doesn't add particular value to the codebase at price of fragmentation, so a good reason to remove it in the refactoring.
However, as you said, is a nitpicky thing, so I wouldn't consider the use of this method a mistake, just it isn't useful for my taste.